rclcpp_components/src/component_manager.cpp

```diff
@@ -46,6 +46,20 @@ ComponentManager::ComponentManager(
   listNodes_srv_ = create_service<ListNodes>(
     "~/_container/list_nodes",
     std::bind(&ComponentManager::OnListNodes, this, _1, _2, _3));
+
+  {
+    rcl_interfaces::msg::ParameterDescriptor desc{};
+    desc.name = "thread_num";
+    desc.type = rcl_interfaces::msg::ParameterType::PARAMETER_INTEGER;
+    desc.description = "Number of thread";
+    rcl_interfaces::msg::IntegerRange range{};
+    range.from_value = 1;
+    range.to_value = std::thread::hardware_concurrency();
+    desc.integer_range.push_back(range);
+    desc.read_only = true;
+    this->declare_parameter(
+      desc.name, static_cast<int>(std::thread::hardware_concurrency()), desc);
```

Comment:
the static_cast isn't needed

Reply:
This prevents the build error as below.
```
/home/daisuke/workspace/rclcpp_ws/install/rclcpp/include/rclcpp/node_impl.hpp:204:15: error: call of overloaded ‘ParameterValue(const unsigned int&)’ is ambiguous
  204 |       rclcpp::ParameterValue(default_value),
```

Reply:
I see, prefer using `int64_t` then, as that's the actual parameter type

Reply:
@ivanpauno fixed at https://github.com/ros2/rclcpp/pull/1708/commits/a6103e20acf800ed217d7d21f046b3a2e60dffb1